### ISSUE TITLE
Mismatch evaluation result on MMLongbench-doc?

### ISSUE THREAD

**Issue by xzwj1699** [MoonshotAI/Kimi-VL#46]
Hi, I tested **Kimi-VL-A3B-Instruct** on **MMLongbench-doc** (score **35.1** in your tech-report), but I only got **15.4** in accuracy.

Here is the full benchmark result:

```
Overall Acc: 0.15428044807149402 | Question Number: 1082
Overall F1-score: 0.16589921170496508 | Question Number: 1082
-----------------------
Single-page | Accuracy: 0.20683385357861422 | Question Number: 494
Cross-page | Accuracy: 0.15256860522989552 | Question Number: 372
Unanswerable | Accuracy: 0.03587443946188341 | Question Number: 223
-----------------------
Evidence Sources: Chart | Accuracy: 0.12579588014981272 | Question Number: 178
Evidence Sources: Table | Accuracy: 0.14588049565113786 | Question Number: 218
Evidence Sources: Pure-text (Plain-text) | Accuracy: 0.19797403282456322 | Question Number: 305
Evidence Sources: Generalized-text (Layout) | Accuracy: 0.22777031791737673 | Question Number: 119
Evidence Sources: Figure | Accuracy: 0.18086752063725747 | Question Number: 304
-----------------------
Document Type: Research report / Introduction | Accuracy: 0.1368436073059361 | Question Number: 292
Document Type: Tutorial/Workshop | Accuracy: 0.29791176569948946 | Question Number: 138
Document Type: Academic paper | Accuracy: 0.09818537130094919 | Question Number: 199
Document Type: Guidebook | Accuracy: 0.14158218125960062 | Question Number: 155
Document Type: Brochure | Accuracy: 0.18046453546453545 | Question Number: 100
Document Type: Administration/Industry file | Accuracy: 0.2509963862088045 | Question Number: 81
Document Type: Financial report | Accuracy: 0.05128205128205128 | Question Number: 117
```

Based on evaluation code from MMLongbench-doc, here is the model loading and evaluation code:

```python
import torch
from PIL import Image
from transformers import AutoModel, AutoTokenizer
from transformers import AutoModelForCausalLM, AutoProcessor

def init_model(cache_path):
    model_path = cache_path if (cache_path is not None and cache_path!="None") else "/root/data/moonshotai/Kimi-VL-A3B-Instruct"

    model = AutoModelForCausalLM.from_pretrained(
        model_path,
        torch_dtype="auto",
        device_map="auto",
        trust_remote_code=True,
        attn_implementation="flash_attention_2",
    )

    processor = AutoProcessor.from_pretrained(model_path, trust_remote_code=True)

    tokenizer = AutoTokenizer.from_pretrained(model_path, trust_remote_code=True)
    model.tokenizer = tokenizer
    model.processor = processor
    return model


def get_response_concat(model, question, image_path_list, max_new_tokens=1024, temperature=1.0):

    if isinstance(image_path_list, list):
        images = [Image.open(path) for path in image_path_list]
        messages = [
            {
                "role": "user",
                "content": [
                    {"type": "image", "image": image_path} for image_path in image_path_list
                ] + [{"type": "text", "text": question}],
            },
        ]
        text = model.processor.apply_chat_template(messages, add_generation_prompt=True, return_tensors="pt")
        inputs = model.processor(images=images, text=text, return_tensors="pt", padding=True, truncation=True).to(model.device)
    else:
        images = [Image.open(image_path_list)]
        messages = [
            {"role": "user", "content": [{"type": "image", "image": image_path_list}, {"type": "text", "text": question}]}
        ]
        text = model.processor.apply_chat_template(messages, add_generation_prompt=True, return_tensors="pt")
        inputs = model.processor(images=images, text=text, return_tensors="pt", padding=True, truncation=True).to(model.device)


    generated_ids = model.generate(**inputs, max_new_tokens=512, temperature=0.2, top_p=1)
    
    generated_ids_trimmed = [
        out_ids[len(in_ids) :] for in_ids, out_ids in zip(inputs.input_ids, generated_ids)
    ]
    response = model.processor.batch_decode(
        generated_ids_trimmed, skip_special_tokens=True, clean_up_tokenization_spaces=False
    )[0]
    return response
```

Some other details:

1. I use **DeepSeek API** for answer extraction instead of **OpenAI**, but I think it does not have much difference;
2. I tried to add " Think step by step" in the question part, but it does not have a significant difference.

Could you please tell me how to reproduce the benchmark result shown in Kimi's tech report? 

Thanks a lot!

**Comment by teowu** [comment]
Please try these fixes:
- temperature 0.0 instead of 1.0; 
- Resize image to long edge as 784 or lower, and limit your input to no more than 64/128 images; 
- The prompt we use is: `Please think step-by-step and answer the question about the document. If the question is not answerable from the document, please answer 'not answerable.'.\nQuestion: `
- I cannot see other mismatches: our eval implementation is also based on VLMEvalKit with the two changes above.
